lib/pages/setup/route_list.dart

```diff
@@ -53,89 +54,167 @@ class _RouteListState extends State<RouteList> {
 
     setState(() {
       widget.routeParams.routes = routes;
-      print('## ${routes.length} routes found');
-
-      widget.routeParams.routes.forEach((r) => routeList.add(RouteListEntry(
-            r.title,
-            r.date,
-            r.totalLength,
-          )));
-      
+      widget.routeParams.routes.forEach((r) => _routeList.add(RouteListEntry(r)));
+      if(routes.length > 0) _title = routes[0].title;
       this._routesCalculated = true;
     });
 
   }
 
+  // TODO add localization or remove if not needed
+  headerText() {
+    String paramTitles = 'Start: ';
+    (_title.length > 20) ? paramTitles += '\n\nDistance: ' : paramTitles += '\nDistance: ';
```

Comment:
When the length of an element is to long flutter allows to specify the overflow for a text widget.